### ISSUE TITLE
[Bug] TfExperiment.test()

### ISSUE THREAD

**Issue by contentis** [delira-dev/delira#125]
**Description**
What happens? What should happen?

**Environment**
* OS: Linux
* Python version: 3.7.2
* `delira` version 0.3.2
* How did you install `delira`? [ pip | source | conda | docker ] source

**Reproduction**
Give us a minimal example to reproduce the error
When testing the trained model using TfExperiment.test() the program crashes due to the closure function not being able to find the criterions and optimizer keys, as the dummy trainer only gets initialized partially.

Can be fixed by initializing the following:
optim_fn = self._optim_builder
losses = training_params.nested_get("criterions")
optimizer_params = training_params.nested_get("optimizer_params")

**Additional context**

**Comment by justusschock** [comment]
Can you give us a small model definition to test this?

**Comment by contentis** [comment]
I uploaded my model to: https://github.com/contentis/ALAD.git but its quite big. Unfortunately it is the only working example I have right now. But this should also happen if yore using a different optimizer than the standard one or habe multiple losses in your dict like:
`
"criterions": {
            'l1': {'mae_': tf.losses.mean_squared_error},
            'l2': {'mse_': tf.losses.absolute_difference},
            'logits': {'ce_': tf.losses.sigmoid_cross_entropy},
`

`
def closure()......
l2 = metrics['l2']
for key, metric_fn in l2.items():
                if isinstance(metric_fn, type):
                l2[key] = metric_fn(f_real.shape)
`

**Comment by justusschock** [comment]
I'll have a look at it, but I can't exactly tell you when
Edit: can you maybe try again on latest master? There've been a couple big fixes during trainer refactoring

**Comment by justusschock** [comment]
I'll close this for now as this should be fixed on latest `master`. feel free to reopen, if the issue persists
